### ISSUE TITLE
Add  message length checks to the Source Interface

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#6297]
## Description

Some production instances have seen an uptick in messages containing only a few characters and no actionable content. This may be attributable to bots, or to UX failures, or a combination of both. Adding a length requirement on initial messages would help in both cases. 
This change should take account of the following:

- the requirement should be applied on initial submissions only. Arbitrarily-short messages should be considered valid in the case of a source-journalist conversation after initial contact
- the requirement should not apply to initial submissions that also include a document, as the actionable content may be in the document
- the length requirement should be displayed on the submission form for initial submissions, to avoid unpleasant surprises for sources.
- rejected messages should result in useful feedback to sources, and should not prevent them from submitting a valid message afterwards
- the minimum length should be configurable by admins in the JI instance config panel, and it should be disabled by default.

## User Research Evidence

Spam reports from instance admins.

**Comment by eloquence** [comment]
One practical concern here is that we don't really have the ability to _reliably_ distinguish initial submissions from returning submitters. 

Previous user interviews suggest that it is quite common for legitimate sources to never use the codename feature, and to simply make a new account every time they want to continue a conversation. Much to the frustration of journalists, who then have to try to piece together who said what.

While those subsequent responses may typically be longer, we don't also know that to be true.

I worry that the end result would be surprising and frustrating when it triggers in the course of a legitimate conversation. The conditional logic when it triggers could make it even more frustrating in some ways, making it opaque to the user why the software sometimes stops them and sometimes doesn't.

**Comment by zenmonkeykstop** [comment]
Ok, but a) that isn't a behavior that we really wanna support, and b) how are said sources actually having a dialog, since they can't read replies? If the comms are one-way, I guess, but they'd still presumably need to identify their past comms somehow, eg. "hey I'm the one who sent you those Evil Corp docs, here are some more"

**Comment by eloquence** [comment]
Of course we'd want sources to come back using their codename, but I think we have to acknowledge that real-world interactions don't necessarily map against our view of how the system should be used - in reality they may _rarely_ do so, and the "coming back under a new codename" behavior may be more common. 

I also don't know that users would necessarily have the mental model to realize that they have to do context-setting to continue a conversation under a new codename. People generally tend to assume software does things for them magically in ways they don't have to worry about.

In short this seems like the kind of change that can easily have side effects we don't want. I'm not necessarily against it with a feature flag, but IMO in that case the character minimum should be visible, and always apply, to avoid confusing users with sometimes showing errors and sometimes not.

**Comment by zenmonkeykstop** [comment]
Down with making it visible - *not* down with it always applying, as that conflicts with potential short responses to replies.

**Comment by eloquence** [comment]
> Arbitrarily-short messages should be considered valid in the case of a source-journalist conversation after initial contact

How would you perform that check? If we set a flag the moment that a journalist reply has been sent, that could work.

Thinking about it more, I think there are two main cases:

1) Same source returning under new codename. They'll always get the minimum character count, clearly advertised, so hopefully they won't be confused.
2) Same source returning under same codename. They'll no longer get the minimum character count the moment they try to type a response, so hopefully they won't be confused/frustrated either.

**Comment by zenmonkeykstop** [comment]
the source's `last_updated` field is null until the first submission, seems like the most straightforward way.

**Comment by eloquence** [comment]
If I understand correctly, then the behavior would be:
1. Source submits a message, minimum character rule applies
2. Source submits another message in the same session, minimum character rule is gone

This seems problematic to me in terms of inconsistent behavior, especially for repeated use (same source returning under new codename, source contacting multiple SDs, or otherwise interacting with the system repeatedly).

IMO it would be preferable to tie disabling the check to whether a journalist reply has ever been sent or received. That would ensure that any initial submissions have the check as a spam/nonsense guard, but if/when journalists engage, we treat it as a conversation.

**Comment by zenmonkeykstop** [comment]
TBH that approach seems *more* inconsistent. Maybe one way to approach this is to think about how to describe it to sources:
- scenario 1: "if your initial submission is just a message, it should contain enough information for a journalist to decide whether to follow up on your tip"
- scenario 2: "Until a journalist replies, you can submit as many messages as you like, but those messages must all exceed a given character limit. After that the limit is no longer imposed."

Obviously I'm being a bit rhetorically sneaky here :) but the first scenario does make more intuitive sense to me, I'd struggle to articulate the second one for sources.

**Comment by eloquence** [comment]
I think they're both inconsistent, which is why I don't really love any of these solutions. My main concern is with the folks who don't follow the expected "happy path", e.g., they may want to start with a short message and then send a file, or we may be dealing with returning users as previously noted.

Between the options that sometimes disable the rule, I tend to favor ones that at least guarantee consistency within a _session_. The message could very simply inform the user of the minimum character rule when it is enabled (and it would of course be indicated below the textarea as well). But I'm curious what others think, as well.

I understand of course that we're trying to determine if this helps with nuisance submissions. Knowing that any solution will have usability tradeoffs, we could make it available as an option, see if people report that it actually helps them, and if not, remove it again from a subsequent release.
